app/authentication/jwt_decoder.py

```diff
@@ -67,16 +162,33 @@ def decrypt_jwt_token(self, token):
                 return self.decode_signed_jwt_token(signed_token)
             else:
                 raise NoTokenException("JWT Missing")
-        except jwt.DecodeError as e:
+        except (jwt.DecodeError, InvalidTag, InternalError, ValueError) as e:
             raise InvalidTokenException(repr(e))
 
+    def __check_jwe_protected_header(self, header):
+        header = self._base64_decode(header).decode()
+        if header.count("alg") == 0:
```

Comment:
as per previous comment, someone putting 'alg' in value element would 'pass' this check but wouldn't be correct (i.e. could still be missing an 'alg' key in the header)